data_model/src/domain.rs

```diff
@@ -30,6 +32,7 @@ pub struct GenesisDomain {
 impl GenesisDomain {
```

Comment:
do we need `GenesisDomain` in the `data_model` or can we move it into `iroha_core`?
Can user submit a transactions containing genesis domain?

Reply:
the same goes for `GenesisAccount`

Reply:
I would prefer they be moved into `iroha_core`.

Reply:
opened #1996